kernel/src/process_standard.rs

```diff
@@ -448,7 +453,10 @@ impl<C: Chip> Process for ProcessStandard<'_, C> {
     }
 
     fn terminate(&self, completion_code: Option<u32>) {
-        if !self.is_running() {
+        if !self.is_running()
+            && self.get_state() != State::Faulted
+            && self.get_state() != State::CredentialsApproved
+        {
```

Comment:
Can you add a comment here explaining why these three states are all part of the early return? I think before it made sense to not terminate a process that isn't running anyways. Now I don't find the logic obvious though.

Reply:
Great catch. I've added a comment.